docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+# Writing Console Apps: A Step by Step Guide
+
+This guide will show you how to use the .NET CLI tooling to build cross-platform console apps incrementally.  It will start with the most basic console app and eventually span multiple projects, including testing.
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
```

Comment:
Can we adopt different file-names for those docs. They are misleading.

I'm not sure that there is such a thing the "console app paradigm".